OpenRA.Mods.Common/Traits/StoresResources.cs

```diff
@@ -49,7 +49,10 @@ public StoresResources(Actor self, StoresResourcesInfo info)
 
 		void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
 		{
+			var capacity = info.Capacity;
+			player.RemoveStorage(capacity);
```

Comment:
In theory the game already removes an actor from the world during owner change (see [Actor.cs#L328](https://github.com/OpenRA/OpenRA/blob/824db72a4cfbd8c6f82bbe60b4bea5c6da327f56/OpenRA.Game/Actor.cs#L328)), so this should already be done from inside the `INotifyRemovedFromWorld.RemovedFromWorld` call or am I missing something here?

Reply:
Looks like you are right, based this pattern on how its implemented in Power trait.